src/VisualStudio/Core/Def/Implementation/ProjectSystem/MiscellaneousFilesWorkspace.cs

```diff
@@ -210,14 +220,32 @@ private void TrackOpenedDocument(uint docCookie, string moniker)
 
         private void Registration_WorkspaceChanged(object sender, EventArgs e)
         {
+            // We may or may not be getting this notification from the foreground thread if another workspace
+            // is raising events on a background. Let's send it back to the UI thread since we can't talk
+            // to the RDT in the background thread. Since this is all asynchronous a bit more asynchrony is fine.
+            if (!_foregroundThreadAffinitization.IsForeground())
+            {
+                ScheduleTask(() => Registration_WorkspaceChanged(sender, e));
```

Comment:
@jasonmalinowski The workspace involved here is the `MiscellaneousFilesWorkspace`?  Is the right fix in that case just to change it's taskQueue to use the ForegroundScheduler the way the the `VisualStudioWorkspace` does?

@jmarolf In the repro, is the problem that in a command line build we don't know what the foreground thread even is because we don't make a WPF dispatcher or something?